godot-core/src/obj/traits.rs

```diff
@@ -46,17 +46,21 @@ where
     fn inherits<U: GodotClass>() -> bool {
         if Self::class_name() == U::class_name() {
             true
-        } else if Self::Base::class_name() == <()>::class_name() {
+        } else if Self::Base::class_name() == <NoBase>::class_name() {
             false
         } else {
             Self::Base::inherits::<U>()
         }
     }
 }
 
-/// Unit impl only exists to represent "no base", and is used for exactly one class: `Object`.
-impl GodotClass for () {
-    type Base = ();
+/// Type representing an object without a base. It is used as the base class for exactly one class: `Object`.
+///
+/// This is an enum without any variants as we should never construct an instance of this class.
+pub enum NoBase {}
```

Comment:
I would shorten the brief doc slightly, since it's shown in module overviews:

```suggestion
/// Type representing the absence of a base class, at the root of the hierarchy.
///
/// `NoBase` is used as the base class for exactly one class: [`Object`][godot::engine::Object].
///
/// This is an enum without any variants, as we should never construct an instance of this class.
pub enum NoBase {}
```